scalding-core/src/main/scala/com/twitter/scalding/ExecutionContext.scala

```diff
@@ -31,6 +36,38 @@ trait ExecutionContext {
   def flowDef: FlowDef
   def mode: Mode
 
+  private def getIdentifierOpt(descriptions: Seq[String]): Option[String] = {
+    if (descriptions.nonEmpty) Some(descriptions.distinct.mkString(", ")) else None
+  }
+
+  private def updateJobName(step: BaseFlowStep[JobConf], descriptions: Seq[String]): Unit = {
+    val conf = step.getConfig
+    getIdentifierOpt(getDesc(step)).foreach(id => {
+      val newJobName = "%s %s".format(conf.getJobName, id)
+      println("Added descriptions to job name: %s".format(newJobName))
```

Comment:
also, this looks like it is changing how all jobs are named. That is somewhat orthogonal to descriptions, right?

We have a lot of tooling that assumes the job names are not changed. Could we instead just add a key to the jobConf?